src/tests/unit_tests/config/test_task_config.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+from langchain import GoogleSerperAPIWrapper
+from langchain.llms import OpenAI
+
+import sherpa_ai.config as cfg
+from sherpa_ai.actions.google_search import GoogleSearch
+from sherpa_ai.config import AgentConfig
+from sherpa_ai.tools import ContextTool, SearchTool
+
+
+def test_all_parameters_parse_successfully():
+    site = "https://www.google.com"
+    input_str = f"Test input. --verbose --gsite {site}"
+
+    parsed, config = AgentConfig.from_input(input_str)
+
+    assert parsed == "Test input."
+    assert config.verbose
+    assert config.gsite == site
+
+
+def test_no_gsite_parses_successfully():
+    input_str = "Test input. --verbose"
+
+    parsed, config = AgentConfig.from_input(input_str)
+
+    assert parsed == "Test input."
+    assert config.verbose
+    assert config.gsite is None
+
+
+def test_parse_args_noise():
+    site = "https://www.google.com"
+    input_str = f"This is an input with -- but--should not be considered --verbose --verbosex --gsite {site} --do-reflect"  # noqa: E501
+
+    with pytest.raises(ValueError):
+        AgentConfig.from_input(input_str)
+
+
+def test_invalid_input_raises_value_error():
```

Comment:
This test is incorrectly named. I think it should be something like `test_gsite_parses_successfully`.

But I'm not sure this test should be here at all. It looks like an integration test, testing the combination of AgentConfig.from_input() and GoogleSearch.config_gsite_query().

Reply:
I think we should change this test to only test GoogleSearch Tool and move it to the tests of tools.

Reply:
I agree, moving this test to a different file and narrowing it to only test GoogleSearchTool is a good next step

Reply:
Cool, I moved it to [src/tests/unit_tests/actions/test_google_search.py](https://github.com/Aggregate-Intellect/sherpa/pull/200/files#diff-ac4c7f8cd7a22d0916dae4bc21bff4690c4590a268c118779ff8bf36018a0479)